sources/packages/api/src/services/disbursement-receipt/disbursement-receipt.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from "@nestjs/common";
+import { Connection } from "typeorm";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { DisbursementReceipt } from "../../database/entities";
+
+/**
+ * Service for disbursement receipts.
+ */
+@Injectable()
+export class DisbursementReceiptService extends RecordDataModelService<DisbursementReceipt> {
+  constructor(private readonly connection: Connection) {
+    super(connection.getRepository(DisbursementReceipt));
+  }
+
+  /**
+   * Insert disbursement receipt record.
+   ** On insertion ignore duplicate records(identified by constraint disbursement_schedule_id_funding_type_unique).
+   * @param disbursementReceipt
+   */
+  async insertDisbursementReceipt(
+    disbursementReceipt: DisbursementReceipt,
+  ): Promise<void> {
+    await this.connection.transaction(async (transactionalEntityManager) => {
+      //Query builder inserts does not cascade insert with relationships.
+      //Cascaded insert can be achieved only through repository.save().
+      //Hence inside a transaction we are using save to persist relations.
+
+      const result = await transactionalEntityManager
+        .createQueryBuilder()
+        .insert()
+        .into(DisbursementReceipt)
+        .values(disbursementReceipt)
+        .orIgnore(
+          "ON CONSTRAINT disbursement_schedule_id_funding_type_unique DO NOTHING",
+        )
+        .execute();
+      if (
```

Comment:
Can we refactor the below code to remove the `[0]`? Something as below:
![image](https://user-images.githubusercontent.com/61259237/174680701-c5e12047-b0fc-4688-9a9c-69e55764d187.png)

Reply:
The second was not changed, can it be changed, please?
![image](https://user-images.githubusercontent.com/61259237/174850507-d0f8b30c-a5ff-4a19-9dcf-0d5c7f213686.png)